app/controllers/rubygems_controller.rb

```diff
@@ -20,6 +20,7 @@ def index
   def show
     @latest_version = @rubygem.versions.most_recent
     @versions       = @rubygem.public_versions(5)
+    render :show_all_yanked if @rubygem.public_versions.count.zero?
```

Comment:
I think `@rubygem.public_versions.none?` can be used here. And better to hide this into `Rubygem` model with some reasonable name so you can call something similar to `@rubygem.yanked?`.

*Is there actual any term for gem with all versions yanked? Does that mean that gem was deleted?*